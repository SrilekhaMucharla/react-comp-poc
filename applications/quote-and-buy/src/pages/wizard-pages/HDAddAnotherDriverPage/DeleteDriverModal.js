import React from 'react';
import PropTypes from 'prop-types';
import { AnalyticsHDModal as HDModal } from '../../../web-analytics';
import * as messages from './HDAddAnotherDriver.messages';

const DeleteDriverModal = ({
    onCancel, onConfirm, onClose, driverName, show, pageMetadata
}) => (
    <HDModal
        webAnalyticsView={{ ...pageMetadata, page_section: messages.removeDriver }}
        webAnalyticsEvent={{ event_action: messages.removeDriver }}
        id="remove-driver-modal"
        show={show}
        headerText={messages.areYouSure}
        cancelLabel={messages.goBack}
        confirmLabel={messages.yes}
        onCancel={onCancel}
        onConfirm={onConfirm}
        onClose={onClose}
    >
        <span>
    Are you sure you want to remove
            {driverName ? <b>{` ${driverName}`}</b> : ' this driver'}
    ?
        </span>
    </HDModal>
);

DeleteDriverModal.propTypes = {
    onCancel: PropTypes.func.isRequired,
    onConfirm: PropTypes.func.isRequired,
    onClose: PropTypes.func.isRequired,
    driverName: PropTypes.string,
    show: PropTypes.bool,
    pageMetadata: PropTypes.shape({
        page_name: PropTypes.string.isRequired,
        page_type: PropTypes.string.isRequired,
        sales_journey_type: PropTypes.string.isRequired
    }).isRequired
};

DeleteDriverModal.defaultProps = {
    driverName: '',
    show: false
};

export default DeleteDriverModal;
