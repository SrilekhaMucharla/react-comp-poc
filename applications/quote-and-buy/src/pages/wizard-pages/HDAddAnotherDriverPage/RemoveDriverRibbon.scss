.remove-driver-ribbon__hide-anim {
    top: 0;
}