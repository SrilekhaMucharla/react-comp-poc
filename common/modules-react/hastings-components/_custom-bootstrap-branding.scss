// Do nothing here

//TODO - move these definitions

///* import only the necessary Bootstrap files */
//@import "../../../node_modules/bootstrap/scss/functions";
//@import "../../../node_modules/bootstrap/scss/variables";
//@import "../../../node_modules/bootstrap/scss/mixins";
//
//@import "./custom-bootstrap-branding-variables";
//
///* import bootstrap to set changes */
//@import "../../../node_modules/bootstrap/scss/bootstrap";

//body {
//  font-family: $font-family-sans-serif;
//}


//.justify-content-md-center {
//  justify-content: center !important;
//}

//.btn-group-toggle {
//  display: unset;
//
//  > .btn:not(:last-child),
//  > .btn-group:not(:last-child) {
//    margin-right: 16px;
//  }
//
//  @include media-breakpoint-up(lg) {
//    > .btn:not(:last-child),
//    > .btn-group:not(:last-child) {
//      margin-right: 22px;
//    }
//  }
//
//  > .btn:not(:last-child):not(.dropdown-toggle),
//  > .btn-group:not(:last-child) > .btn {
//    @include border-radius(0.625rem);
//  }
//
//  > .btn:not(:first-child),
//  > .btn-group:not(:first-child) > .btn {
//    @include border-radius(0.625rem);
//  }
//
//
//  @include media-breakpoint-up(lg) {
//    > .btn {
//      width: 189px;
//      height: 60px;
//    }
//  }
//
//  > .btn {
//    display: inline-flex;
//    font-weight: 600;
//    font-size: 1.125rem;
//    margin-bottom: 30px;
//
//    > .justify-content-md-center {
//      align-items: center;
//      flex-direction: column;
//      display: flex;
//      width: 100%;
//    }
//  }
//
//  > .btn-primary:hover {
//    border-color: unset;
//  }
//}

//.horizontal-line {
//  width: 100%;
//  height: 1px;
//  border-radius: 6px;
//  background-color: $blue-transparent-30
//}