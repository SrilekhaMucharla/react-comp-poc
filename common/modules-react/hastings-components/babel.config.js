module.exports = require('./../../../common/build-tools/gw-build-config-babel');
