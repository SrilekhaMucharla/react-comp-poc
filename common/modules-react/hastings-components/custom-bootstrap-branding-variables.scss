/* make the customizations */
$theme-colors: (
        "primary": #fff,
);

$border-radius: 0.625rem;
$font-family-sans-serif: 'TTNormsPro-Bold', "Helvetica Neue", BlinkMacSystemFont, "Segoe UI", Roboto, Arial, "Noto Sans", sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji";

$border-color: #727272;
$border-color-active: #585757;
$border-solid-xl: 1px solid $border-color;
$border-solid-active-button: 1px solid $border-color-active;
$background-color-active: #DDDDDD;
$secondaryButtonBGColor: #FFFFFF;
$headerTextColor: #333333;
$primaryButtonBGColor: #404040;
$tertiaryButtonBGColor: #4F4F4F;
$secondaryTextColor: $secondaryButtonBGColor;
$regularButtonBorderColor: #9C9C9C;

/** Custom Hastings colors, based on theme.scss from main App **/
$alternateBG: #f7f7f9;
$appTheme: #011831;
$pageBG: #ffffff;
$regNumberBackgroundColor: #FBE700;

$genericTitleBoldFontFace: $font-family-sans-serif;
$genericFontFace: 'TTNormsPro';
$genericNormalFontFace: 'TTNormsPro-Normal';
$genericRegularFontFace: 'TTNormsPro-Regular';
$fontStyleNormal: normal;
$sectionHeaderColor: #02213e;
$secondaryLink: #AFBDD1;

$commonCompoBGColor: #01355a;
$commonCompoBGColorLightBlue2: #f3f8ff;
$commonCompoBGColorBlue: #00448c;
$commonCompoBGColorRed:#a90d30;
$commonCompoBGColorWhite: #ffffff;
$commonCompoBGColorBlueLight:#0069cc;
$commonCompoBGColorGreen:#0e838a;
$commonCompoTextColorBlueDark : #011831;
$commonCompoTextColorBlueGray: #60759e;
$commonCompoTextColorBlueLight:#0069cc;
$commonCompoTextColorBlueGrayLight: #afbdd1;
$commonCompoTextColorGraySecondary:#4c6172;
$commonCompoTextColor: #ffffff;
$commonCompoBorderColor:  #60759e;
$commonCompoBorderColorRed:#e10930;
$commonCompoBorderColorGreen:#00a39e;
$commonCompoBorderColorGreenDark:#0e838a;
$commonCompoBorderColorBlueLight: #0085ff;
$commonCompoBorderColorBlueGrayLight: #afbdd1;

$customizeQuoteBtnColor: #0E838A;
$customizeQuoteSecondaryBtnColor: rgba(14, 131, 138, 0.2);
$customizeQuoteSecondaryBtnBorderColor: #00A39E;
$customizeQuotePopupBackgroundColor: rgba(1, 24, 49, 0.95);

$purple-secondary: #a24593;

$breakpoint-mobile: 375px;
$breakpoint-tablet: 768px;
$breakpoint-desktop: 1140px;

/* Exclusively for usage in styled components */
:export {
  appTheme: $appTheme;
  secondaryTextColor: $secondaryTextColor;
  commonCompoBGColor: $commonCompoBGColor;
  commonCompoBGColorWhite: $commonCompoBGColorWhite;
  commonCompoBGColorLightBlue2: $commonCompoBGColorLightBlue2;
  commonCompoTextColor: $commonCompoTextColor;
  commonCompoTextColorBlueDark: $commonCompoTextColorBlueDark;
  tertiaryTextColor: $commonCompoTextColorBlueGrayLight;
  genericFontFace: $genericFontFace;
  genericTitleBoldFontFace: $genericTitleBoldFontFace;
}