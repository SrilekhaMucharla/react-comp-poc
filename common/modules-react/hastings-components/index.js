import HDInteractiveCard from './components/HDInteractiveCard/HDInteractiveCard';
import HDInteractiveCardRefactor from './components/HDInteractiveCardRefactor/HDInteractiveCardRefactor';
import HDTextInput from './components/HDTextInput/HDTextInput';
import HDTextInputRefactor from './components/HDTextInputRefactor/HDTextInputRefactor';
import HDButton from './components/HDButton/HDButton';
import HDButtonRefactor from './components/HDButtonRefactor/HDButtonRefactor';
import HDInfoCard from './components/HDInfoCard/HDInfoCard';
import HDInfoCardRefactor from './components/HDInfoCardRefactor/HDInfoCardRefactor';
import HDQuoteInfo from './components/HDQuoteInfo/HDQuoteInfo';
import HDQuoteInfoRefactor from './components/HDQuoteInfoRefactor/HDQuoteInfoRefactor';
import HDQuoteInfoWarning from './components/HDQuoteInfoWarning/HDQuoteInfoWarning';
import HDLabel from './components/HDLabel/HDLabel';
import HDLabelRefactor from './components/HDLabelRefactor/HDLabelRefactor';
import HDStepper from './components/HDStepper/HDStepper';
import HDToggleButtonGroup from './components/HDToggleButtonGroup/HDToggleButtonGroup';
import HDToggleButtonGroupRefactor from './components/HDToggleButtonGroupRefactor/HDToggleButtonGroupRefactor';
import HDClickableCard from './components/HDClickableCard/HDClickableCard';
import HDDropdownList from './components/HDDropdownList/HDDropdownList';
import HDImageRadioButton from './components/HDImageRadioButton/HDImageRadioButton';
import HDButtonDashed from './components/HDButtonDashed/HDButtonDashed';
import HDDataCard from './components/HDDataCard/HDDataCard';
import HDForm from './components/HDForm/HDForm';
import HDWarningModal from './components/HDWarningModal/HDWarningModal';
import HDOverlayPopup from './components/HDOverlayPopup/HDOverlayPopup';
import HDModal from './components/HDModal/HDModal';
import HDAsyncSelect from './components/HDDropdownList/HDAsyncSelect';
import HDInformationToolTip from './components/HDInformationToolTip/HDInformationToolTip';
import HDAlert from './components/HDAlert/HDAlert';
import HDAlertRefactor from './components/HDAlertRefactor/HDAlertRefactor';
import HDDatePicker from './components/HDDatePicker/HDDatePicker';
import HDDatePickerRefactor from './components/HDDatePickerRefactor/HDDatePickerRefactor';
import HDTable from './components/HDTable/HDTable';
import HDAccordion from './components/HDAccordion/HDAccordion';
import HDAccordionRefactor from './components/HDAccordionRefactor/HDAccordionRefactor';
import HDQuoteTable from './components/HDQuoteTable/HDQuoteTable';
import HDQuoteDownload from './components/HDQuoteDownload/HDQuoteDownload';
import HDQuoteDownloadRefactor from './components/HDQuoteDownloadRefactor/HDQuoteDownloadRefactor';
import HDPlaceholderWithHeader from './components/HDPlaceholderWithHeader/HDPlaceholderWithHeader';
import HDContainer from './components/HDContainer/HDContainer';
import HDInfoBox from './components/HDInfoBox/HDInfoBox';
import HDTickListItem from './components/HDTickListItem/HDTickListItem';
import * as yup from './yup';
import HDSwitch from './components/HDSwitch/HDSwitch';
import HDBubble from './components/HDBubble/HDBubble';
import HDCompletedCardInfo from './components/HDCompletedCardInfo/HDCompletedCardInfo';
import HDRibbon from './components/HDRibbon/HDRibbon';
import HDFullscreenLoader from './components/HDFullscreenLoader/HDFullscreenLoader';
import HDSpinner from './components/HDSpinner/HDSpinner';
import HDLoader from './components/HDLoader/HDLoader';
import HDCustomerReviews from './components/HDCustomerReviews/HDCustomerReviews';
import HDCustomerReviewsWidget from './components/HDCustomerReviews/HDCustomerReviewsWidget';
import HDPaymentBreakdown from './components/HDPaymentBreakdown/HDPaymentBreakdown';
import HDToast from './components/HDToast/HDToast';
import HDCheckbox from './components/HDCheckbox/HDCheckbox';
import HDOverlay from './components/HDOverlay/HDOverlay';
import HDPriceTable from './components/HDPriceTable/HDPriceTable';
import HDVehicleSimpleDetails from './components/HDVehicleDetails/HDVehicleSimpleDetails';
import HDRadioButtonList from './components/HDRadioButtonList/HDRadioButtonList';
import HDCheckboxButtonList from './components/HDCheckboxButtonList/HDCheckboxButtonList';
import HDPolicySelect from './components/HDPolicySelect/HDPolicySelect';

export {
    HDTextInput,
    HDTextInputRefactor,
    HDButton,
    HDButtonRefactor,
    HDLabel,
    HDLabelRefactor,
    HDInfoCard,
    HDInfoCardRefactor,
    HDToggleButtonGroup,
    HDToggleButtonGroupRefactor,
    HDClickableCard,
    HDDropdownList,
    HDInteractiveCard,
    HDInteractiveCardRefactor,
    HDImageRadioButton,
    HDButtonDashed,
    HDStepper,
    HDDataCard,
    HDForm,
    HDWarningModal,
    HDModal,
    HDAsyncSelect,
    HDOverlayPopup,
    HDInformationToolTip,
    HDAlert,
    HDAlertRefactor,
    HDDatePicker,
    HDDatePickerRefactor,
    HDTable,
    HDAccordion,
    HDAccordionRefactor,
    HDQuoteTable,
    HDQuoteInfo,
    HDQuoteInfoRefactor,
    HDQuoteInfoWarning,
    HDQuoteDownload,
    HDQuoteDownloadRefactor,
    yup,
    HDPlaceholderWithHeader,
    HDTickListItem,
    HDContainer,
    HDBubble,
    HDInfoBox,
    HDSwitch,
    HDLoader,
    HDCompletedCardInfo,
    HDRibbon,
    HDFullscreenLoader,
    HDSpinner,
    HDCustomerReviews,
    HDCustomerReviewsWidget,
    HDPaymentBreakdown,
    HDToast,
    HDCheckbox,
    HDOverlay,
    HDPriceTable,
    HDVehicleSimpleDetails,
    HDRadioButtonList,
    HDCheckboxButtonList,
    HDPolicySelect
};
