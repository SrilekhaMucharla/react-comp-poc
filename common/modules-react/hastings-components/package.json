{
  "name": "hastings-components",
  "version": "1.0.0",
  "description": "Hastings React Components Library for QnB",
  "main": "index.js",
  "author": "Hastings",
  "private": true,
  "dependencies": {
    "@babel/core": "^7.11.6",
    "@babel/preset-env": "^7.11.5",
    "babel-jest": "^26.3.0",
    "bootstrap": "^4.5.2",
    "formik": "^2.1.7",
    "react-bootstrap": "^1.3.0",
    "react-datepicker": "2.0.0",
    "react-input-mask": "^2.0.4",
    "react-intl": "^2.9.0",
    "react-lottie": "^1.2.3",
    "react-select": "^2.4.4",
    "react-test-renderer": "^16.13.1",
    "styled-components": "^5.2.0",
    "yup": "^0.29.3"
  },
  "peerDependencies": {
    "react": "16.x",
    "react-dom": "16.x",
    "prop-types": "15.x",
    "lodash": "4.x",
    "classnames": "^3.x"
  },
  "devDependencies": {
    "@babel/polyfill": "^7.12.1",
    "@babel/preset-react": "^7.10.4",
    "enzyme": "^3.11.0",
    "enzyme-adapter-react-16": "^1.15.5",
    "identity-obj-proxy": "^3.0.0",
    "jest": "^26.4.2"
  },
  "scripts": {
    "test": "node ./../../build-tools/gw-build-testing/bin/runTest.js",
    "test:oneFile": "node ./../../build-tools/gw-build-testing/bin/runTest.js --oneFile",
    "test:watch": "jest --watchAll"
  }
}
