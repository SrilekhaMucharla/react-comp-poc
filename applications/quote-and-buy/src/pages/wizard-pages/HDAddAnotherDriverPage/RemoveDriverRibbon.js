import React, { useState, useEffect } from 'react';
import { useSelector } from 'react-redux';
import * as _ from 'lodash';
import { HDRibbon } from 'hastings-components';
import { useLocation, useHistory } from 'react-router-dom';
import PropTypes from 'prop-types';
import DeleteDriverModal from './DeleteDriverModal';
import useAnotherDriver from '../__helpers__/useAnotherDriver';
import routes from '../../../routes/WizardRouter/RouteConst';
import * as messages from './HDAddAnotherDriver.messages';
import EventEmmiter from '../../../EventHandler/event';

const RemoveDriverRibbon = ({ pageMetadata }) => {
    const [hideRibbon, setHideRibbon] = useState(false);
    const history = useHistory();
    const location = useLocation();

    const moveRibbonHandler = (event) => {
        setHideRibbon(event.hide);
    };
    EventEmmiter.subscribe('change', moveRibbonHandler);

    const [driverIndex, isAnotherDriver, driverFixedId] = useAnotherDriver(location);
    const [showDeleteModal, setShowDeleteModal] = useState(false);

    const drivers = useSelector((state) => state.wizardState.data.submissionVM.lobData.privateCar.coverables.drivers.value);
    const editDriverIndex = drivers
            && drivers.length && !!driverFixedId && drivers.findIndex((driver) => driver.fixedId === driverFixedId) !== -1
        ? drivers.findIndex((driver) => driver.fixedId === driverFixedId)
        : driverIndex;

    const onDelete = () => setShowDeleteModal(true);

    const hideDeleteModal = () => setShowDeleteModal(false);

    const onDeleteConfirm = () => {
        hideDeleteModal();
        _.pullAt(drivers, editDriverIndex);
        history.push(routes.ADD_ANOTHER_DRIVER);
    };

    const firstName = _.get(drivers, `${editDriverIndex}.person.firstName`, null);
    const lastName = _.get(drivers, `${editDriverIndex}.person.lastName`, null);
    const driverName = firstName && lastName ? ` ${firstName} ${lastName}` : null;

    const hideRibbonCustomClass = (isHidden) => {
        if (isHidden) return 'remove-driver-ribbon__hide-anim';
        return '';
    };

    useEffect(() => {
        return () => {
            EventEmmiter.unsubscribe('change');
        };
    }, []);

    return (
        <>
            <HDRibbon
                text={driverName}
                actionText={isAnotherDriver ? messages.removeDriver : null}
                onClick={onDelete}
                className={`remove-driver-ribbon${hideRibbonCustomClass(hideRibbon)}`} />
            <DeleteDriverModal
                pageMetadata={pageMetadata}
                show={showDeleteModal}
                onCancel={hideDeleteModal}
                onConfirm={() => onDeleteConfirm()}
                onClose={hideDeleteModal}
                driverName={driverName} />
        </>
    );
};

RemoveDriverRibbon.propTypes = {
    pageMetadata: PropTypes.shape({
        page_name: PropTypes.string.isRequired,
        page_type: PropTypes.string.isRequired,
        sales_journey_type: PropTypes.string.isRequired
    }).isRequired,
};

export default RemoveDriverRibbon;
