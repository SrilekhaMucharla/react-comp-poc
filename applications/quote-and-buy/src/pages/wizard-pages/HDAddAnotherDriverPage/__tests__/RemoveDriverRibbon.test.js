import React from 'react';
import { shallow, mount } from 'enzyme';
import { HDRibbon } from 'hastings-components';
import RemoveDriverRibbon from '../RemoveDriverRibbon';
import DeleteDriverModal from '../DeleteDriverModal';

// given
const mockedPush = jest.fn();
const isPolicyHolder = false;
const mockedRouterState = { isPolicyHolder };
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useHistory: () => ({
        push: mockedPush
    }),
    useLocation: () => ({
        state: mockedRouterState
    })
}));

const mockedDrivers = [
    { firstName: 'David', lastName: 'Bowie' },
    { firstName: 'Johnny', lastName: 'Cash' }];
jest.mock('react-redux', () => ({
    ...jest.requireActual('react-redux'),
    useSelector: () => mockedDrivers
}));

const mockedSetState = jest.fn();
jest.mock('react', () => ({
    ...jest.requireActual('react'),
    useState: (init) => [init, mockedSetState]
}));

jest.mock('react-dom', () => ({
    ...jest.requireActual('react-dom'),
    createPortal: (element) => element
}));

describe('RemoveDriverRibbon', () => {
    // given
    const getShallowWrapper = () => shallow(<RemoveDriverRibbon />);
    const getMountWrapper = () => mount(<RemoveDriverRibbon />);

    it('should render correctly and match the snapshot', () => {
        // given
        const wrapper = getShallowWrapper();
        // then
        expect(wrapper).toMatchSnapshot();
    });

    it('should contain HDRibbon', () => {
        // given
        const wrapper = getShallowWrapper();
        // then
        expect(wrapper.find(HDRibbon)).toHaveLength(1);
    });
    it('should contain DeleteDriverModal', () => {
        // given
        const wrapper = getShallowWrapper();
        // then
        expect(wrapper.find(DeleteDriverModal)).toHaveLength(1);
    });

    it('should call onDelete function', () => {
        // given
        const wrapper = getMountWrapper();
        // when
        wrapper.find('[data-testid="action-button"]').simulate('click');
        // then
        expect(mockedSetState).toHaveBeenCalledWith(true);
    });

    it('should hide modal on confirm', async () => {
        // given
        const wrapper = getMountWrapper();
        // when
        wrapper.find('[data-testid="action-button"]').simulate('click');
        // then
        await act(async () => wrapper.update());
        // then
        const modal = wrapper.find(DeleteDriverModal);
        await act(async () => modal.props().onConfirm());
        await act(async () => wrapper.update());
        // then
        expect(modal.prop('show')).toBeFalsy();
    });
});
