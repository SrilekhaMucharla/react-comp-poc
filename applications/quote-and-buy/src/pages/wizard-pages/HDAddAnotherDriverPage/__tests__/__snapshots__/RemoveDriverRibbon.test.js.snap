// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveDriverRibbon should render correctly and match the snapshot 1`] = `
<Fragment>
  <HDRibbon
    actionText="Remove driver"
    className="remove-driver-ribbon"
    onClick={[Function]}
    onKeyDown={[Function]}
    tabIndex={0}
    text={null}
  />
  <DeleteDriverModal
    driverName={null}
    onCancel={[Function]}
    onClose={[Function]}
    onConfirm={[Function]}
    show={false}
  />
</Fragment>
`;
